Matrix.java

```diff
@@ -0,0 +1,210 @@
+class Matrix
+{
+    final double[][] a;
+    
+    Matrix(double[][] A)
+    {
+	this(A,0,0);
+    }
+    Matrix(double[][] A, int m, int n)
```

Comment:
Now that we're starting to implement functions, can we make sure to write a basic Javadoc for each one. It can literally be the single sentence from the API documentation that Dr. Fenwick provided.